File: CS50 Assignments Source Codes/MarioPy/mario.py
from cs50 import get_int

# Prompt user for height using modified while loop, equivalent to do while in C

while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Iterate through every row
for i in range(height):

    # For each row, create the spaces for the left side pyramid
    for j in range(height - i, 1, -1):
        print(" ", end="")

    # For each row, add a hash
    for k in range(i + 1):
        print("#", end="")

    # Print the spaces in middle that will split the pyramids
    for l in range(2):
        print(" ", end="")

    # Print hashes on right side of spaces
    for m in range(i + 1):
        print("#", end="")
    print("")

File: CS50 Assignments Source Codes/MarioC/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8); // Ask user to input height over and over again until they choose one between 1 and 8

    // For every row
    for (int i = 0; i < n; i++)
    {

        // For spaces in each row
        for (int j = n - i; j > 1; j--)
        {
            printf(" ");
        }

        // For each row, add hashes after spaces
        for (int k = 0; k < i + 1; k++)
        {
            printf("#");
        }


        // For each row, add two spaces after the hashes
        for (int l = 0; l < 2; l++)
        {
            printf(" ");
        }


        // For each row, add hashes after the two spaces
        for (int m = 0; m < i + 1; m++)
        {
            printf("#");
        }

        printf("\n");
    }

}